### PR TITLE
WIP: Timeout solve

### PR DESCRIPTION
**Pull request by soapy1**

### CODE REVIEW

File: conda/core/solve.py [hunk 1]
```diff
@@ -25,18 +27,34 @@
...
 from ..models.version import VersionOrder
 from ..resolve import Resolve
+from ..common.signals import raise_timeout
+
 
 log = getLogger(__name__)
 
 
+@contextmanager
+def solver_timeout(time):
```
Comment (soapy1) [inline conda/core/solve.py]: unfortunately this approach does not work for windows or python2

Comment (msarahan) [reply conda/core/solve.py]: on the bright side, we don't need to worry about supporting python 2 anymore!

Comment (soapy1) [reply conda/core/solve.py]: could maybe use something like https://dreamix.eu/blog/webothers/timeout-function-in-python-3 instead

**Comment by travishathaway** [comment]
This pull request has become stale and is therefore being closed. Please re-open if you would like to continue development on this.

**Pull request closed without merging**